### PR TITLE
Fixes #449: Adds ngrx structure for media wall customization

### LINKED ISSUE #449: Ngrx Store for media wall customization
Based on ngrx/store, build up necessary reducers, actions and models for media wall customization.

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Adds ngrx structure for media wall customization
- Customization for -
1.  Wall Header
2. Wall Background
3. Wall Cards

**Closes #449 **

### CODE REVIEW

**Comment by Achint08** [comment]
Open for review @mariobehling @djmgit  @kavithaenair @hemantjadon @singhpratyush @SKrPl.

File: src/app/models/media-wall-custom.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/**
+ * @function defaultColumnCount
+ * Checks the current screen size and returns the default column count
+ *
+ * @return string
+ */
+
+export function defaultColumnCount(): string {
```
Comment (hemantjadon) [inline src/app/models/media-wall-custom.ts]: Why such approach?
Why aren't you simply depending on flexbox this is not good approach, laying out UI by JS.

File: src/app/actions/media-wall-custom.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * enum object for all of this group's action types.
+ *
+ * The 'actionTypeCheck' utility function coerces strings into string
+ * literal types and runs a simple check to guarantee all
+ * action types in the application are unique.
+ */
+export const ActionTypes = {
+	WALL_HEADER_CHANGE: '[Media Wall] Media Wall Header Change',
+	WALL_BACKGROUND_CHANGE: '[Media Wall] Media Wall Background Change',
+	WALL_GRID_LAYOUT_CHANGE: '[Media Wall] Media Wall Grid Layout Change',
```
Comment (hemantjadon) [inline src/app/actions/media-wall-custom.ts]: What is this property used for? What do you mean by layout change? We have only one layout?

File: src/app/actions/media-wall-custom.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	WALL_CARD_CHANGE: '[Media Wall] Media Wall Card Change'
```
Comment (hemantjadon) [inline src/app/actions/media-wall-custom.ts]: Also this?

File: src/app/models/media-wall-custom.ts (hunk 1 above)
Comment (Achint08) [inline src/app/models/media-wall-custom.ts]: Using flex-box creates an issue of vertical gap. If we use flex boxes, gaps are produced depending on the maximum height of a card in a row. Using column count can make item align vertically without any gaps.
That is why column count would be a good approach for this.

**Comment by Achint08** [comment]
@hemantjadon Made the changes, Please review now. :)

**Pull request merged**